homeassistant/components/octoprint/__init__.py

```diff
@@ -250,15 +267,17 @@ def get(self, endpoint):
             self.available = False
             return None
 
-    def update(self, sensor_type, end_point, group, tool=None):
+    def update(self, sensor_type, end_point, group, secondary_group=None, tool=None):
         """Return the value for sensor_type from the provided endpoint."""
         response = self.get(end_point)
         if response is not None:
-            return get_value_from_json(response, sensor_type, group, tool)
+            return get_value_from_json(
+                response, sensor_type, group, secondary_group, tool
+            )
         return response
 
 
-def get_value_from_json(json_dict, sensor_type, group, tool):
+def get_value_from_json(json_dict, sensor_type, group, secondary_group, tool):
```

Comment:
The device specific interface code, I/O and data parsing, should be extracted to a standalone library published on pypi.

https://developers.home-assistant.io/docs/en/creating_component_code_review.html#4-communication-with-devices-services